src/ray/gcs/gcs_server/gcs_placement_group_manager.cc

```diff
@@ -815,7 +815,26 @@ void GcsPlacementGroupManager::OnNodeDead(const NodeID &node_id) {
       // TODO(ffbin): If we have a placement group bundle that requires a unique resource
       // (for example gpu resource when there's only one gpu node), this can postpone
       // creating until a node with the resources is added. we will solve it in next pr.
-      if (iter->second->GetState() != rpc::PlacementGroupTableData::RESCHEDULING) {
+
+      // check to make sure the placement group shouldn't be in PENDING or REMOVED state
+      RAY_CHECK_NE(iter->second->GetState(), rpc::PlacementGroupTableData::PENDING)
+          << "Placement group that is pending rescheduling shouldn't be in PENDING "
+             "state. placement_group_id="
```

Comment:
.WithField(iter->second->GetPlacementGroupID()).WithField(node_id) << "PENDING placement group should have no scheduled bundles on the dead node"